graph/pull_v2.go

```diff
@@ -76,6 +76,9 @@ func (p *v2Puller) pullV2Repository(tag string) (err error) {
 	broadcaster.Add(p.config.OutStream)
 	if found {
 		// Another pull of the same repository is already taking place; just wait for it to finish
+		if dryRun {
+			broadcaster.Write(p.sf.FormatStatus(tag, "!!! Another pull of the same image is already running, dry-run is not possible unless you restart the Docker daemon !!!"))
```

Comment:
Also, consider what happens if someone tries to do `docker pull myimage` and `docker pull -dry-run myimage` at the same time. If the dry-run pull starts first, the normal pull may attach to that broadcaster, and the pull will never happen.

In dry-run mode, this function should probably avoid calling `poolAdd` and `poolRemoveWithError` at all, to avoid being mistaken for an actual pull. It could pass `p.config.OutStream` to `pullV2Tag` in this case, instead of `broadcaster`.